src/commands/commit/withClient/index.js

```diff
@@ -5,9 +5,18 @@ import chalk from 'chalk'
 import isHookCreated from '@utils/isHookCreated'
 import configurationVault from '@utils/configurationVault'
 import { type Answers } from '../prompts'
+import isGitRepository from '@utils/isGitRepository'
 
 const withClient = async (answers: Answers): Promise<void> => {
   try {
+    if(!isGitRepository()) {
+      return console.log(
+        chalk.red(
+          "\nError: Seems that you're trying to commit outside a git repository \n" +
+          'Please navigate to a git repository to use `gitmoji` \n'
+        )
+      )
+    }
```

Comment:
```suggestion
```

Reply:
This is what's causing the error to display in a weird way, we can just remove this:

https://github.com/carloscuesta/gitmoji-cli/blob/68bc2a460f12f642fde5a0c3d5543fd055788b00/src/utils/isHookCreated.js#L16-L19

Reply:
The only issue I see is that it will not be a controlled error. 
We won't know inside the catch block what has actually happened.

Another thing to note - if we don't bail out with the helper, we ask the user for input knowing they are kind of doomed to fail. 🙈

Reply:
> Another thing to note - if we don't bail out with the helper, we ask the user for input knowing they are kind of doomed to fail. 🙈

Yep that's the downside of this approach, however I would say it should be a very unfrequent use-case and given that situation probably makes sense to do something that can improve the overall experience of the errors? 

> The only issue I see is that it will not be a controlled error.
> We won't know inside the catch block what has actually happened.

We will rely on `git` to tell the user what happened, I think this is better than wrapping specific use-cases only

Reply:
> We will rely on git to tell the user what happened, I think this is better than wrapping specific use-cases only
Yeah exactly my train of thought as well. I would just expose the actual errors to the user instead of masking them.

Reply:
Good point!

I reverted the changes with your recommendations. Now it breaks and the error is handled by the top level catch block.

That original error handling was nice, since it told you the git command it was going to execute before it failed. The problem now is that it fails in another git command and the message no longer makes sense. See the attached image for reference.

The only solution with this approach would be to make that error message more generic, which might be too generic 😞 

![image](https://github.com/carloscuesta/gitmoji-cli/assets/8035145/7d6a415c-e9eb-47f6-8583-07b6cfe9b59d)

Reply:
Both are fine!

I don't think that loosing the `escapedCommand` thingy is too much of a loss, if we improve the overall error handling

So feel free to go with any of the options!

Reply:
Great! Thanks.
I removed the initial approach and just made the error handling more generic.
Hope it helps.

Reply:
Thanks for making the required changes! Ideally the files you touched are also converted to TS 🙏

Reply:
I'm a bit weary of migrating to Typescript file by file. 
I might be doing it wrong, but as soon as I change one file, I have to basically change all the depending files and tests in a cascading effect.
I'll keep an eye on the repo and lend a hand if I figure out how you're doing it.